galaxy_skeleton2.R

```diff
@@ -31,44 +32,51 @@ raw_file <- readLines(con = "?")
 
 # What do you need to replace the two question marks with in order to extract
 # the first two letters?
-substr(x = raw_file, start = ?, stop = ?)
+substr(x = raw_file, start = 1, stop = 2) # extracts the first two letters 
 
 # The next step is then to find out *which* line starts with "--", and pick out
 # the first one. This can be done in a nice little pipe, where you have to fill
 # out the question marks and the missing function names:
 L <- 
-  (substr(x = raw_file, start = ?, stop = ?) == "?") %>% 
-  function_that_returns_the_index_of_all_TRUES %>% 
-  function_that_picks_out_the_minimum_value
+  (substr(x = raw_file, start = 1, stop = 2) == "--") %>% # extract the first two letters that needs to be "--"
+  which() %>% # function that returns the index of all TRUES
+  min() # function that picks out the minimum value (for where the line starts with "--")
+
+# The output is 14, which means it is the 14th line in the text file that starts with "--"
+# Hence, it is line 14 that separates the column names and the rest of the data table 
 
 # Save the variable descriptions (i.e. the information in lines 1:(L-2)) in a
 # text-file for future reference using the cat()-function. The first argument is
 # the information that we want to print out. In order to get each element in the
 # "raw_file"-vector on a separate line we also provide the sep-argument, where
 # we put the "end-of-line"-character "\n". We also need to come up with a file
 # name. Replace the question marks:
-cat(?, sep = "\n", file = "?")
 
-# Extract the variable names (i.e. line (L-1)), store the names in a vector.
+cat(raw_file[1:(L-2)], # the info we want to print out, i.e. the variable description lines in raw_file
+    sep = "\n", # to get each element in raw_file on a separate line
+    file = "galaxy_variable_descriptions.txt") # file name 
 
+# Extract the variable names (i.e. line (L-1)), store the names in a vector.
 # This is a little bit dirty. We want to *split* the string in raw_data[L-1]
```

Comment:
I would recommend to remove the existing comments and instead add your own ones.